src/condition_variable.hpp

```diff
@@ -212,8 +219,16 @@ class condition_variable_t
                 timeout.tv_sec++;
                 timeout.tv_nsec -= 1000000000;
             }
-
+#ifdef ZMQ_HAVE_OSX
+            rc = pthread_cond_timedwait_relative_np (
```

Comment:
Is this available on all versions of OSX? There have been many issues with OSX and clocks over the years, see:
https://github.com/zeromq/libzmq/issues/2537
https://github.com/zeromq/libzmq/pull/2538
https://github.com/zeromq/libzmq/issues/1351
https://github.com/zeromq/libzmq/pull/1354

Reply:
I found `pthread_cond_timedwait_relative_np` recommended at least 6 years ago for macOS.

For Android, looks like that was the wrong check, and I should just be checking for r21 or not:
https://stackoverflow.com/questions/44580542/ndk-builder-r15-finds-neither-have-pthread-cond-timedwait-monotonic-nor-pthread

Reply:
It's referenced in https://opensource.apple.com/source/Libc/Libc-166/pthreads.subproj/pthread_cond.c, which is literally OS X 10.0. So yeah I'd say it's probably fine.

Reply:
Great, thanks for checking

Reply:
Android build fixed. The clang format build is still erroring, but not on my changes.

Reply:
Great, thanks for your contribution, merged